openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +751,104 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def get_context_var_changes():
+    """get context var changes."""
+
+    houdini_vars_to_update = {}
+
+    project_settings = get_current_project_settings()
+    houdini_vars_settings = \
+        project_settings["houdini"]["general"]["update_houdini_var_context"]
+
+    if not houdini_vars_settings["enabled"]:
+        return houdini_vars_to_update
+
+    houdini_vars = houdini_vars_settings["houdini_vars"]
+
+    # No vars specified - nothing to do
+    if not houdini_vars:
+        return houdini_vars_to_update
+
+    # Get Template data
+    template_data = get_current_context_template_data()
+
+    # Set Houdini Vars
+    for item in houdini_vars:
+        # For consistency reasons we always force all vars to be uppercase
+        item["var"] = item["var"].upper()
+
+        # get and resolve template in value
+        item_value = StringTemplate.format_template(
+            item["value"],
+            template_data
+        )
+
+        if item["var"] == "JOB" and item_value == "":
+            # sync $JOB to $HIP if $JOB is empty
+            item_value = os.environ["HIP"]
+
+        if item["is_directory"]:
+            item_value = item_value.replace("\\", "/")
+
+        current_value = hou.hscript("echo -n `${}`".format(item["var"]))[0]
+
+        # sync both environment variables.
+        # because houdini doesn't do that by default
+        # on opening new files
+        os.environ[item["var"]] = current_value
+
+        if current_value != item_value:
+            houdini_vars_to_update.update(
+                {
+                    item["var"]: (
+                        current_value, item_value, item["is_directory"]
+                    )
+                }
+            )
```

Comment:
```suggestion
        # For consistency reasons we always force all vars to be uppercase
        var = item["var"].upper()

        # get and resolve template in value
        item_value = StringTemplate.format_template(
            item["value"],
            template_data
        )

        if var == "JOB" and item_value == "":
            # sync $JOB to $HIP if $JOB is empty
            item_value = os.environ["HIP"]

        if item["is_directory"]:
            item_value = item_value.replace("\\", "/")

        current_value = hou.hscript("echo -n `${}`".format(var))[0]

        # sync both environment variables.
        # because houdini doesn't do that by default
        # on opening new files
        os.environ[var] = current_value

        if current_value != item_value:
            houdini_vars_to_update[var] = (
                current_value, item_value, item["is_directory"]
            )
```

Reply:
I did notice another issue aside of my above suggestion. We're changing `os.environ[var]` in a function that should only **get** the changes we might want to make. I know that's because in this case you'd also need to return those values that have non-changed values.

How about changing the `get_context_var_changes` to have an extra argument? For example:
```python
def get_context_var_changes(differences_only=True):
```

Then `update_houdini_vars_context` would call it: `get_context_var_changes(differences_only=False)` and would set the `os.environ[]` value even for non-changed values but continues to do the rest for changed values.

Thoughts?

Reply:
I have two ideas 
1. we can just move it and update function will eventually update and sync them with the new value 
2. sync old values before the popup

Reply:
I will just remove it.